File: python/series/series.py
def slices(series, length):
    series_length = len(series)

    if length > series_length or length <= 0 or not series:
        raise ValueError("The series or slice length is invalid!")

    return [''.join(series[i:i + length]) for i in range(series_length - length + 1)]

File: javascript/series/series.js
export class Series {
  constructor(series) {
    this._digits = [...series].map(Number);
  }

  get digits() {
    return this._digits;
  }

  slices(length) {
    if (length > this._digits.length) throw new Error('Slice size is too big.');

    return this._digits.reduce((slices, _, index) => {
      if (index + length <= this._digits.length) slices.push(this._digits.slice(index, index + length));
      return slices;
    }, []);
  }
}
